debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java

```diff
@@ -626,7 +628,8 @@ protected void handleLobWrite(LogMinerEventRow row) {
         }
 
         if (row.getRedoSql() != null) {
-            addToTransaction(row.getTransactionId(), row, () -> new LobWriteEvent(row, parseLobWriteSql(row.getRedoSql())));
+            final ParsedLobWriteSql parsed = parseLobWriteSql(row.getRedoSql());
+            addToTransaction(row.getTransactionId(), row, () -> new LobWriteEvent(row, parsed.data, parsed.offset, parsed.length));
```

Comment:
Do we need to explicitly always invoke the parse on the `LOB_WRITE` event?  I specifically moved a number of these into the actual Lambda function instead so that if `addToTransaction` decides to discard the event that we don't unnecessarily parse the SQL text.  So unless there is a specific reason to do this always, I would suggest we adjust these lines to look like the following:

```suggestion
            addToTransaction(row.getTransactionId(), row, () -> {
              final ParsedLobWriteSql parsed = parseLobWriteSql(row.getRedoSql());
              return new LobWriteEvent(row, parsed.data, parsed.offset, parsed.length));
            });
```

Reply:
done